pkg/spacebinding/spacebindinglister_test.go

```diff
@@ -235,4 +235,207 @@ func TestNewSpaceBindingLister(t *testing.T) {
+			switch spaceName {
+			case spaceA.GetName():
+				fmt.Println("return binding for SpaceA")
+				return []toolchainv1alpha1.SpaceBinding{
+					*spacebinding.NewSpaceBinding(spaceAMur, spaceA, spaceA.Name, spacebinding.WithRole("admin")),
+				}, nil
+			case spaceB.GetName():
+				fmt.Println("return binding for SpaceB")
+				return []toolchainv1alpha1.SpaceBinding{
+					*spacebinding.NewSpaceBinding(spaceBMur, spaceB, spaceB.Name, spacebinding.WithRole("viewer")),
+				}, nil
+			case spaceC.GetName():
+				fmt.Println("return binding for SpaceC")
+				return []toolchainv1alpha1.SpaceBinding{
+					*spacebinding.NewSpaceBinding(spaceCMur, spaceC, spaceC.Name, spacebinding.WithRole("maintainer")),
+				}, nil
+			case spaceD.GetName():
+				fmt.Println("return binding for SpaceD")
+				return []toolchainv1alpha1.SpaceBinding{
+					*spacebinding.NewSpaceBinding(spaceDMur, spaceD, spaceD.Name, spacebinding.WithRole("admin")),
+				}, nil
+			case spaceE.GetName():
+				fmt.Println("return no binding for SpaceE")
+				return []toolchainv1alpha1.SpaceBinding{}, nil
+			case spaceF.GetName():
+				fmt.Println("return binding for SpaceF")
+				return []toolchainv1alpha1.SpaceBinding{
+					*spacebinding.NewSpaceBinding(spaceFMur, spaceF, spaceF.Name, spacebinding.WithRole("viewer")),
+				}, nil
+			default:
+				return nil, nil
+			}
+		}
+
+		tests := map[string]struct {
+			space                 *toolchainv1alpha1.Space
+			getSpaceFunc          func(spaceName string) (*toolchainv1alpha1.Space, error)
+			listSpaceBindingsFunc func(spaceName string) ([]toolchainv1alpha1.SpaceBinding, error)
+			expectedSpaceBindings []toolchainv1alpha1.SpaceBinding
+			expectedErr           string
+		}{
+			"for Space-A": {
+				space:                 spaceA,
+				getSpaceFunc:          getSpaceFunc,
+				listSpaceBindingsFunc: listSpaceBindingFunc,
+				expectedSpaceBindings: []toolchainv1alpha1.SpaceBinding{
+					// we expect to have one spacebinding for the root space
+					*spacebinding.NewSpaceBinding(spaceAMur, spaceA, spaceA.Name, spacebinding.WithRole("admin")),
+				},
+			},
+			"for Space-B": {
+				space:                 spaceB,
+				getSpaceFunc:          getSpaceFunc,
+				listSpaceBindingsFunc: listSpaceBindingFunc,
+				expectedSpaceBindings: []toolchainv1alpha1.SpaceBinding{
+					// only expect space binding form space-B
+					*spacebinding.NewSpaceBinding(spaceBMur, spaceB, spaceB.Name, spacebinding.WithRole("viewer")),
+				},
+			},
+			"for Space-C": {
+				space:                 spaceC,
+				getSpaceFunc:          getSpaceFunc,
+				listSpaceBindingsFunc: listSpaceBindingFunc,
+				expectedSpaceBindings: []toolchainv1alpha1.SpaceBinding{
+					// we expect spacebinding inherited from space-B
+					*spacebinding.NewSpaceBinding(spaceBMur, spaceB, spaceB.Name, spacebinding.WithRole("viewer")),
+					// and specific one from space-C (which overrides the one from space-B even if it has the same role)
```

Comment:
it doesn't override anything - I guess that it's copy-paste from the previous test case
```suggestion
					// and the one from space-C
```

Reply:
removed